x-pack/plugins/apm/public/components/shared/Links/apm/transaction_detail_link.tsx

```diff
@@ -41,7 +43,8 @@ export function TransactionDetailLink({
         transactionId,
         transactionName,
         transactionType,
-        ...persistedFilters,
+        ...(latencyAggregationType ? { latencyAggregationType } : {}),
+        ...pickKeys(urlParams as APMQueryParams, ...persistedFilters),
```

Comment:
Perhaps outside the scope of this PR but I think it's confusing that we have two layers of persisted filters (one local for TransactionDetailLink, and the shared one in used by `getAPMHref`). 

What about removing `PERSISTENT_APM_PARAMS` from `getAPMHref` like:
```ts
export function getAPMHref({
  basePath,
  path = '',
  ...rest
}: {
  basePath: IBasePath;
  path?: string;
} & (
  | {
      search: string;
    }
  | {
      query: APMQueryParams;
    }
)) {
  return url.format({
    pathname: basePath.prepend(`/app/apm${path}`),
    search: 'search' in rest ? rest.search : fromQuery(rest.query),
  });
}
```

and then importing `PERSISTENT_APM_PARAMS` where needed?

Reply:
@sqren Might be good to limit the amount of changes we are making here?